doc/cygnus-ngsi/installation_and_administration_guide/grouping_rules.md

```diff
@@ -0,0 +1,50 @@
+#<a name="top"></a>Grouping Rules
+The file `grouping_rules.conf` can be instantiated from a template given in the Cygnus repository, `conf/grouping_rules.conf.template`.
+
+Grouping rules allow changing the notified <b>FIWARE service path<b> and the concatenation of <b>entity ID<b> and <b>entity type<b> (this is called the destination).
+Such rules are just a Json within a configuration file detailing how notified <b>FIWARE service path<b> must be replaced by alternative <b>FIWARE service path<b>.
+
+Grouping rules are writing in Json formal. The following code is just an example:
+
+```
+{
+    "grouping_rules": [
+        {
+            "id": 1,
+            "fields": [
+                ...
+            ],
+            "regex": "...",
+            "destination": "...",
+            "fiware_service_path": "..."
+        },
+        ...
+    ]
+}
+```
+
+When a notification is sent to Cygnus, a special Flume interceptor called `NGSIGroupingInterceptor` intercepts the plain `Event`'s generated by `NGSIRestHandler`, parses the `Event`'s body and replace the <b>FIWARE service path<b> of the `Event` with the notified one in the Grouping Rule. If `fields` it not null the modify is done with the notified fields such as <b>entityId<b>, <b>entityType> or <servicePath>. Once finished, the modified version of the notification is put into the channel buy means of a `NGSIEvent`. Explained fields of a Grouping Rule:
+
+* <b>id</b>: A unique unsigned integer-based identifier.
+* <b>fields</b>: These are the fields that will be concatenated for regular expression matching. The available dictionary of fields for concatenation is "entityId", "entityType" and "servicePath". The order of these fields is important since the concatenation is made from left to right.
+* <b>regex</b>: Java-like regular expression to be applied on the concatenated fields. Special characters like '\' must be escaped ('\' is escaped as "\\\\").
+* <b>destination</b>: Name of the HDFS file or CKAN resource where the data will be effectively persisted. In the case of MySQL, Mongo and STH Comet this sufixes the table/collection name.
+* <b>fiware\_service\_path</b>: New `fiware-servicePath` replacing the notified one. The sinks will translate this into the name of the HDFS folder or CKAN package where the above destination entity will be placed. In the case of MySQL, Mongo and STH Comet this prefixes the table/collection name. It must start with `/` or the whole rule will be discarded.
+
```

Comment:
An "Important notes" section is required, containing this paragrah (tailored to grouping rules):

Please observe the name mappings definition is global to all the sinks, at NGSIRestHandler, but then the application is local to the sink, depending on the enable_name_mappings parameter. Thus, if any of your sinks is going to take advantage of the mappings, simply leave blank the mappings configuration file in the REST handler. That will avoid unnecessary interception and iterations on the mappings.